codejail/jail_code.py

```diff
@@ -210,10 +211,9 @@ def jail_code(command, code=None, files=None, extra_files=None, argv=None,
         if user:
             # Run as the specified user
             cmd.extend(['sudo', '-u', user])
+            cmd.extend('{}="{}"'.format(var, env[var]) for var in env)
             rm_cmd.extend(['sudo', '-u', user])
 
-        # Point TMPDIR at our temp directory.
-        cmd.extend(['TMPDIR=tmp'])
```

Comment:
@nedbat: Is there a reason this does `'TMPDIR=tmp'` rather than `'TMPDIR={}'.format(tmptmp)`?  I left it the way it was, because I'm thinking there might be some chrooting going on somewhere that would break tmptmp inside the jail.